shark_turbine/kernel/wave/promotion.py

```diff
@@ -38,3 +40,15 @@ def promote_node(node: Read | Write, address_space: IndexSymbol):
         )
         allocate_node.add_to_graph(node.graph)
         apply_promotion_pattern(node, allocate_node)
+
+
+def promote_placeholders(graph: CapturedTrace):
+    for node in graph.get_root_graph().nodes:
+        custom = get_custom(node)
+        if isinstance(custom, Read) or isinstance(custom, Write):
```

Comment:
Seems like `apply_promotion_pattern` that is used by `promote_node` only handles `Read`, are we just adding `Write` since we plan on adding support for it later?

Reply:
Yes, in fact this is what Jakub was originally working on. Promoting the writes so that we have nice global write patterns. With Ivan's mapping construct, I think we could implement this.

Reply:
sounds good!